attalos/imgtxt_algorithms/approaches/negsampling.py

```diff
@@ -71,55 +72,58 @@ def meanlogsig(predictions, truth):
     def __init__(self, wv_model, datasets, **kwargs):
         self.wv_model = wv_model
         self.one_hot = OneHot(datasets, valid_vocab=wv_model.vocab)
+        word_counts = NegativeSampler.get_wordcount_from_datasets(datasets, self.one_hot)
```

Comment:
added comment to clarify in code. Word_counts is a numpy array of length equal to the vocab. The contents of the array are how often each word occurs in datsets.